### ISSUE TITLE
PLEASE READ: Not triggering on incoming messages? Add a Personal Automation due to recent Spotify share link changes

### ISSUE THREAD

**Issue by iannotian** [iannotian/imessage2spotify-serverless#6]
## TL;DR - What all users should do

If/when/before iMessage2Spotify stops detecting Spotify links, add a Personal Automation that runs iMessage2Spotify for messages containing `spotify.link`. You should also keep the old automation for `open.spotify.com/track` since some clients still send those kinds of links.

See video: https://user-images.githubusercontent.com/1900909/226857840-2818b720-cd3f-4968-937c-f286a0a056b8.mov

## Issue description

Share links were updated by Spotify. I've seen the new links when sharing with the iOS client, but the Mac client seems to still use the old links. This will break iMessage2Spotify, so don't expect it to work right now. It will still work if you receive links from a Mac user.

Old links used to look like:
`https://open.spotify.com/track/4GtTxyApv3YlkR0jaxarAN`

New links look like this:
`https://spotify.link/cDQJVG7Pmyb`

ChatGPT explains it pretty well:

> The new Spotify links appear to use a different, shorter format compared to the old ones. The old links follow a pattern that includes the domain (https://open.spotify.com/), the type of content (track, album, playlist, etc.), and a unique identifier (4GtTxyApv3YlkR0jaxarAN). The new links include the domain (https://spotify.link/) and a shorter, seemingly random alphanumeric code (cDQJVG7Pmyb).
> 
> From the information provided, it is not possible to derive the older-style links directly from the new-style links. It seems that the new links use a different encoding scheme or an internal mapping system to redirect users to the appropriate content. Without knowing the exact mapping or conversion method used by Spotify, it's not possible to derive the old-style links from the new-style ones.

~~I'll see if the API allows me to search for a song with those new style links. If so, I'll also have to update the app so that it can correctly process the new link.~~ It's fixed! No updates are necessary for users; just add a Personal Automation to run this Shortcut when you receive a message with the text `spotify.link`.

**Comment by iannotian** [comment]
Reopening for visibility.
